content/geometry/SegmentIntersection.h

```diff
@@ -16,31 +16,24 @@ Products of three coordinates are used in intermediate steps so watch out for ov
  * Status: Well tested with fuzz-test and with Kattis problem intersection.
  * Usage:
  * vector<P> inter = segInter(s1,e1,s2,e2);
- * if (inter.size()==1)
+ * if (sz(inter)==1)
  *   cout << "segments intersect at " << inter[0] << endl;
  */
 #pragma once
 
 #include "Point.h"
 #include "onSegment.h"
 
-template <class P>
-bool segInterProper(P a, P b, P c, P d, P& out) {
-	double oa = c.cross(d, a), ob = c.cross(d, b),
-	       oc = a.cross(b, c), od = a.cross(b, d);
-	if (sgn(oa) * sgn(ob) < 0 && sgn(oc) * sgn(od) < 0) {
-		out = (a * ob - b * oa) / (ob - oa);
-		return true;
-	}
-	return false;
-}
 template<class P> vector<P> segInter(P a, P b, P c, P d) {
-	P out;
-	if (segInterProper(a, b, c, d, out)) return {out};
+	auto oa = c.cross(d, a), ob = c.cross(d, b),
+	     oc = a.cross(b, c), od = a.cross(b, d);
+	// Returns if only one intersection exists
```

Comment:
This comment is not quite true; the one-intersection case can also occur if one of the endpoints lines on the other line. Rather, this finds intersections of open, non-parallel line segments.